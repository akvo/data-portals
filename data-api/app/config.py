from typing import Any, Dict, Optional

from pydantic import BaseSettings, PostgresDsn, validator


class Settings(BaseSettings):
    POSTGRES_HOST: str
    POSTGRES_USER: str
    POSTGRES_PASSWORD: str
    POSTGRES_DB: str
    DATABASE_URI: Optional[PostgresDsn] = None

    @validator("DATABASE_URI", pre=True)
    def assemble_db_connection(cls, v: Optional[str], values: Dict[str, Any]) -> Any:
        if isinstance(v, str):  # pragma: no cover
            return v
        return PostgresDsn.build(
            scheme="postgresql",
            user=values.get("POSTGRES_USER"),
            password=values.get("POSTGRES_PASSWORD"),
            host=values.get("POSTGRES_HOST"),
            path=f"/{values.get('POSTGRES_DB') or ''}",
        )
