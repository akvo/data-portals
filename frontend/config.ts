export const INTERNAL_API_DOMAIN = process.env.INTERNAL_API_DOMAIN || ''

export const API_PATH = process.env.API_PATH || ''

export const API_URL = INTERNAL_API_DOMAIN + API_PATH
