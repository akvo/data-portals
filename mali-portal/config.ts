export const API_HOST = process.env.API_HOST

export const DATA_ENDPOINT = process.env.DATA_ENDPOINT

export const DATA_API_ENDPOINT = process.env.DATA_API_ENDPOINT

export const DEFAULT_MAP_VIEWPORT = {
  latitude: 17.65,
  longitude: -4.15,
  zoom: 4.4,
}
