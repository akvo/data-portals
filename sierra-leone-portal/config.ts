export const API_HOST = process.env.API_HOST

export const DATA_ENDPOINT = process.env.DATA_ENDPOINT

export const DATA_API_ENDPOINT = process.env.DATA_API_ENDPOINT

export const DEFAULT_MAP_VIEWPORT = {
  latitude: 8.4,
  longitude: -11.8,
  zoom: 6.5,
}
