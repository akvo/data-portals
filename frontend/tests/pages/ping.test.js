import React from 'react'
import { render } from '@testing-library/react'
import Ping, { getServerSideProps } from '../../pages/ping'
import { server, rest } from '../mocks/server'
import { INTERNAL_API_DOMAIN } from '../../config'

describe('Ping page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Ping ping="pong!" />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('should call backend endpoint', async () => {
    server.use(
      rest.get(INTERNAL_API_DOMAIN + '/ping', (_, res, ctx) => {
        return res(ctx.json({ ping: 'pong!' }))
      })
    )
    const response = await getServerSideProps()
    expect(response).toEqual({ props: { ping: 'pong!' } })
  })
})
