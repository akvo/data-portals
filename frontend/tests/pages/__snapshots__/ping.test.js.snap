// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ping page matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    pong!
  </h1>
  <a
    href="/"
  >
    Home
  </a>
  <p>
    postgres
  </p>
</DocumentFragment>
`;
